src/website/shared/models/cve.py

```diff
@@ -153,9 +153,96 @@ class Impact(models.Model):
 class Metric(models.Model):
     """Class representing an impact information related to a CVE record."""
 
+    class Scopes(models.TextChoices):
+        UNCHANGED = (
+            "UNCHANGED",
+            _("UNCHANGED"),
+        )
+        CHANGED = (
+            "CHANGED",
+            _("CHANGED"),
+        )
+
+    class AttackVectors(models.TextChoices):
+        PHYSICAL = ("PHYSICAL", _("PHYSICAL"))
+        LOCAL = (
+            "LOCAL",
+            _("LOCAL"),
+        )
+        ADJACENT_NETWORK = (
+            "ADJACENT_NETWORK",
+            _("ADJACENT_NETWORK"),
+        )
+        NETWORK = ("NETWORK", _("NETWORK"))
+
+    class Severity(models.TextChoices):
+        NONE = ("NONE", _("NONE"))
+        LOW = ("LOW", _("LOW"))
+        MEDIUM = ("MEDIUM", _("MEDIUM"))
+        HIGH = ("HIGH", _("HIGH"))
+        CRITICAL = ("CRITICAL", _("CRITICAL"))
```

Comment:
This looks extremely tedious to me. Why can't we

```python
class ChoiceEnum(Enum):
    @classmethod
    def choices(cls):
        return [(choice.name, _(str(choice.value))) for choice in cls]

    def __str__(self):
        return self.name

class Scopes(ChoiceEnum):
    UNCHANGED = "UNCHANGED"
    CHANGED = "CHANGED"

class AttackVectors(ChoiceEnum):
    PHYSICAL = "PHYSICAL"
    LOCAL = "LOCAL"
    ADJACENT_NETWORK = "ADJACENT_NETWORK"
    NETWORK = "NETWORK"

class Severity(ChoiceEnum):
    NONE = "NONE"
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    CRITICAL = "CRITICAL"
```

And also I'm not sure I fully understand why we need a `TextChoice` and can't do `IntegerChoice` protobuf-style. I guess we'd have to do a mapping on ingestion, but that seems low-overhead.

Reply:
Your proposal does not support built-in gettext translation. And these strings _should_ be translated.

Reply:
Doesn't `return [(choice.name, _(str(choice.value))) for choice in cls]` cut it?

Reply:
This is not statically discoverable.

Reply:
Tools needs to mark the callsites for `_("")`